File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/season/TvSeasonPaths.java
package dev.reuss.tmdb.domain.tv.season;

import dev.reuss.tmdb.value.id.TvSeasonId;
import dev.reuss.tmdb.value.id.TvShowId;

final class TvSeasonPaths {

    private static final String TV = "/tv";

    private TvSeasonPaths() {
    }

    static String details(TvShowId tvShowId, int seasonNumber) {
        return season(tvShowId, seasonNumber);
    }

    static String aggregateCredits(TvShowId tvShowId, int seasonNumber) {
        return season(tvShowId, seasonNumber) + "/aggregate_credits";
    }

    static String changes(TvShowId tvShowId, int seasonNumber) {
        return season(tvShowId, seasonNumber) + "/changes";
    }

    static String changes(TvSeasonId seasonId) {
        return TV + "/season/" + seasonId.asString() + "/changes";
    }

    static String credits(TvShowId tvShowId, int seasonNumber) {
        return season(tvShowId, seasonNumber) + "/credits";
    }

    static String externalIds(TvShowId tvShowId, int seasonNumber) {
        return season(tvShowId, seasonNumber) + "/external_ids";
    }

    static String images(TvShowId tvShowId, int seasonNumber) {
        return season(tvShowId, seasonNumber) + "/images";
    }

    static String translations(TvShowId tvShowId, int seasonNumber) {
        return season(tvShowId, seasonNumber) + "/translations";
    }

    static String videos(TvShowId tvShowId, int seasonNumber) {
        return season(tvShowId, seasonNumber) + "/videos";
    }

    static String watchProviders(TvShowId tvShowId, int seasonNumber) {
        return season(tvShowId, seasonNumber) + "/watch/providers";
    }

    private static String season(TvShowId tvShowId, int seasonNumber) {
        return TV + "/" + tvShowId.asString() + "/season/" + seasonNumber;
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/season/TvSeasonPaths.kt
package dev.reuss.tmdb.domain.tv.season

import dev.reuss.tmdb.core.path.tmdbPath
import dev.reuss.tmdb.value.id.TvSeasonId
import dev.reuss.tmdb.value.id.TvShowId

internal object TvSeasonPaths {

    fun details(
        tvShowId: TvShowId,
        seasonNumber: Int
    ): String =
        tmdbPath("tv", tvShowId, "season", seasonNumber)

    fun aggregateCredits(
        tvShowId: TvShowId,
        seasonNumber: Int
    ): String =
        tmdbPath(
            "tv",
            tvShowId,
            "season",
            seasonNumber,
            "aggregate_credits"
        )

    fun changes(seasonId: TvSeasonId): String =
        tmdbPath("tv", "season", seasonId, "changes")

    fun credits(
        tvShowId: TvShowId,
        seasonNumber: Int
    ): String =
        tmdbPath("tv", tvShowId, "season", seasonNumber, "credits")

    fun externalIds(
        tvShowId: TvShowId,
        seasonNumber: Int
    ): String =
        tmdbPath("tv", tvShowId, "season", seasonNumber, "external_ids")

    fun images(
        tvShowId: TvShowId,
        seasonNumber: Int
    ): String =
        tmdbPath("tv", tvShowId, "season", seasonNumber, "images")

    fun translations(
        tvShowId: TvShowId,
        seasonNumber: Int
    ): String =
        tmdbPath("tv", tvShowId, "season", seasonNumber, "translations")

    fun videos(
        tvShowId: TvShowId,
        seasonNumber: Int
    ): String =
        tmdbPath("tv", tvShowId, "season", seasonNumber, "videos")

    fun watchProviders(
        tvShowId: TvShowId,
        seasonNumber: Int
    ): String =
        tmdbPath(
            "tv",
            tvShowId,
            "season",
            seasonNumber,
            "watch",
            "providers"
        )
}